### PR TITLE
Emscripten support + Strikeout and Underline values

### PR DESCRIPTION
**Pull request by krogueintel**
Add Emscripten build support + metric values for underlines and strikeouts.

### CODE REVIEW

File: src/fastuidraw/text/font_freetype.cpp [hunk 1]
```diff
@@ -900,6 +902,24 @@ compute_font_metrics_from_face(FT_Face in_face, FontMetrics &out_metrics)
...
         .descender(in_face->descender)
-        .units_per_EM(in_face->units_per_EM);
+        .units_per_EM(in_face->units_per_EM)
+        .underline_position(in_face->underline_position)
+        .underline_thickness(in_face->underline_thickness);
+
+      TT_OS2* os2;
+
+      os2 = static_cast<TT_OS2*>(FT_Get_Sfnt_Table(in_face, ft_sfnt_os2));
+      if (os2)
```
Comment (capr) [inline src/fastuidraw/text/font_freetype.cpp]: did you mean `if(!os2)` ?

Comment (kevin-rogovin) [reply src/fastuidraw/text/font_freetype.cpp]: oops, yes.

**Comment by kevin-rogovin** [comment]
I made a mess of the commits... Sighs.. will open a new pr.

**Comment by krogueintel** [comment]
New PR up, which fixes the typo. Sighs. thanks for catching it.

**Pull request closed without merging**